Automattic/jetpack
# Protect: Add user connection gate for applicable actions

#### Description
With the introduction of the paid UI as part of the Scan in Protect integration, actions are now available to the users (fix and ignore threats) that will require a user connection. This PR establishes a user connection gate for these particular actions.

#### Changes proposed in this Pull Request:

* Adds a `user-connection-gate` component that checks for the presence of a user connection and responds accordingly
* Adds a `user-connection-needed-modal` component that displays in the absence of a user connection only when the process of fixing or ignoring a threat is initiated
* Clarifies what actions require a connection, why we need it, and how to go about connecting
* Redirects to the Protect admin page and presents the appropriate modal after the connection is established

#### Other information:

#### Jetpack product discussion


- 1201069996155224-as-1203180130348952

#### Does this pull request change what data or activity we track or use?


- No

#### Testing instructions:






* Start up a Jurassic Ninja instance w/ the Jetpack Beta plugin, a Scan level plan, and this Protect branch installed
* Install a known threat
* Activate Protect, select the Free connection (site-only) and allow the initial report to generate
* Run JPDB to ensure your connection is site-only, and downgrade if necessary
* Return to Protect admin page, select Fix all threats (#), Ignore threat, and Fix threat and verify that for each you are presented with a prompt for a user connection
* Proceed to connect your user and be redirected back to the Protect admin page
* Verify that each of the above buttons now present the following:
   * Fix all threats (#) - a modal prompt for server credentials
   * Fix threat - a modal prompt for server credentials
   * Ignore threat - a modal prompt to confirm or cancel the ignore threat action
* Ensure there are no regressions in functionality

#### Screenshots
<img width="614" alt="Screen Shot 2022-10-18 at 07 05 18" src="https://user-images.githubusercontent.com/43220201/196452591-b6c32ca4-ca21-4b27-92ca-a199f0b83567.png">
